homeassistant/components/jellyfin/entity.py

```diff
@@ -32,3 +34,44 @@ def __init__(
             name=self.coordinator.server_name,
             sw_version=self.coordinator.server_version,
         )
+
+
+class JellyfinClientEntity(JellyfinEntity):
```

Comment:
Would it make sense to split even further and split the `JellyfinEntity` into a `JellyfinEntity` and a `JellyfinServerEntity`? Because now the `JellyfinEntity` requires an entity description, which results in that we only create a small `RemoteEntityDescription` for the key, and even though it works, we don't really have a good reason to create one.

(If we do this, I would recommend doing this in a preliminary PR so we can keep the scope on this one nice and tight :) )

Reply:
It would also help with always putting in the session id in the unique_id, so adding more entities for the client in the future would be easier

Reply:
> Would it make sense to split even further and split the JellyfinEntity into a JellyfinEntity and a JellyfinServerEntity?

Not sure, I'm not an expert in this area. I'm assuming there was a reason `JellyfinMediaPlayer` was based off of the server-based `JellyfinEntity` so my default stance was not to change that.

IMO it makes sense having the server be the base: all entities ultimately are linked (at least conceptually) to the Jellyfin server, with some also being linked to individual clients.

> Because now the JellyfinEntity requires an entity description, which results in that we only create a small RemoteEntityDescription for the key, and even though it works, we don't really have a good reason to create one.

My understanding is we want to pass the descriptor to help determine the entity type (e.g. `MediaPlayerEntityDescription` for `JellyfinMediaPlayer` vs. `RemoteEntityDescription` for `JellyfinRemote`). So that information needs to be passed in or set somewhere.

Reply:
Oh but that doesn't matter here, because your `JellyfinRemote` inherits `RemoteEntity` and the mediaplayer `MediaPlayer` entity. We usually use entity descriptions to create 1 generic entity class and change the data and appearance of that entity with injecting an entity description, but that is not the case with the remote

Reply:
Ah, that makes sense.

So in this case you're saying split out the server entity so that `JellyfinSensor` can still pass in its description without requiring the clients to pass in a description, allowing the client entity to set `coordinator`/`_attr_unique_id`/`_attr_device_info` directly?

IDK, I feel like the description being required isn't necessarily a client vs. server thing and there could be client-based entities in the future that want to pass in more than a key in the description :thinking: Not sure, I try not to change up existing code when I don't have a solid understanding of all the minutia.

I can get a PR up changing all of this, but it'll be a little until I have the time to make the changes + test.

Reply:
Oh i forgot to mention, but i also opened a PR to simplify jellyfin and I have a local branch for the base classes

Reply:
Ah that's great, I think that'll be much easier than playing telephone since I'm not super familiar with the HA codebase :slightly_smiling_face: I'll give your PR a try when I have a minute.

Reply:
Yea currently it just simplifies HA since it's currently set up for extension, but it's not extended. So my PR simplifies to make the code better